File: exs/mundo_1/c#/022.cs
/*
Desafio 022

Problema: Crie um programa que leia o nome completo de uma pessoa
          e mostre:
              - O nome com todas as letras maiúsculas e minúsculas.
              - Quantas letras ao todo(sem considerar espaços)
              - Quantas letras tem o primeiro nome.

Resolução do problema:
*/

using System;

class io {
    public static string input(string text) {
        Console.Write(text);
        return Console.ReadLine();
    }
    
    public static void print(object data, string end="\n") {
        string text = data.ToString();

        Console.Write($"{text}{end}");
    }
}


class Program {
  public static void Main () {
    string nome = io.input("Digite seu nome completo: ");

    string nomeMaiusculo = nome.ToUpper();
    string nomeMinusculo = nome.ToLower();
    int qntdLetrasNomeCompleto = nome
      .Replace(" ", "")
      .Length;
    int qntdLetrasPrimeiroNome = nome
      .Split(' ')[0]
      .Length;


    io.print($"Nome maíusculo: {nomeMaiusculo}");
    io.print($"Nome minúsculo: {nomeMinusculo}");
    io.print($"Quantidade de letras que o nome completo tem: {qntdLetrasNomeCompleto}");
    io.print($"Quantidade de letras que apenas o primeiro nome tem: {qntdLetrasPrimeiroNome}");
  }
}

File: exs/mundo_1/rust/022.rs
/*
Desafio 022

Problema: Crie um programa que leia o nome completo de uma pessoa
          e mostre:
              - O nome com todas as letras maiúsculas e minúsculas.
              - Quantas letras ao todo(sem considerar espaços)
              - Quantas letras tem o primeiro nome.
*/

fn main() {
    let mut input = String::new();

    println!("Digite o nome completo: ");
    std::io::stdin().read_line(&mut input).unwrap();

    input = input.trim().to_string();

    let minusculo = input.to_lowercase();
    let maiusculo = input.to_uppercase();
    let contador_letras = input.replace(" ", "").len();
    let primeiro_nome = input.split(" ").nth(0).unwrap();

    println!("Nome completo: {}", maiusculo);
    println!("Nome completo: {}", minusculo);
    println!("Quantidade de letras: {}", contador_letras);
    println!("Quantidade de letras do primeiro nome: {}", primeiro_nome);
}
